src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy

```diff
@@ -564,17 +580,21 @@ class DevTask extends AbstractFeatureTask {
                 // - start server
                 util.restartServer();
                 return true;
-            } else if (installFeatures) {
-                if (generateFeatures) {
-                    // Increment generate features on build dependency change
-                    ProjectConnection gradleConnection = initGradleProjectConnection();
-                    BuildLauncher gradleBuildLauncher = gradleConnection.newBuild();
-                    runGradleTask(gradleBuildLauncher, 'compileJava', 'processResources'); // ensure class files exist
-                    Collection<String> javaSourceClassPaths = getJavaSourceClassPaths();
-                    libertyGenerateFeatures(javaSourceClassPaths, false);
-                    libertyCreate(); // need to run create in order to copy generated config file to target
-                }
+            } else if (installFeatures && !compileProject) {
                 libertyInstallFeature();
+            } else if (compileProject && generateFeatures) { // generate features if compile dependencies have been modified
+                ProjectConnection gradleConnection = initGradleProjectConnection();
+                BuildLauncher gradleBuildLauncher = gradleConnection.newBuild();
+                runGradleTask(gradleBuildLauncher, 'compileJava', 'processResources'); // ensure class files exist
```

Comment:
For Maven, we are not compiling on compilation dependency change correct? It seems like this would be a new thing for Gradle. Should we keep it consistent between the two plugins?

Reply:
Also updated to no longer call `create` and `installFeature` since that will be handled by the rest of the dev mode flow and to match the behaviour in ci.maven.